cachi2/core/package_managers/rpm/main.py

```diff
@@ -462,3 +480,50 @@ def _generate_repofiles(
 
             with open(repo_file_path, "w") as f:
                 repofile.write(f)
+
+
+def _get_ssl_context(options: Optional[dict] = None) -> ssl.SSLContext:
+
+    ssl_ctx = ssl.create_default_context()
+
+    if options is None:
+        log.debug("returning default ssl context.")
+        return ssl_ctx
```

Comment:
We always want options set in this function.

Reply:
I had _SSLOptions set as default to _SSLOptions() before and you asked me to remove it and have it default to None.  If options.ssl can be None, it makes sense that the function can accept None.  If you don't want it to accept None, I think the most reasonable way to accomplish it is to have _ExtraOptions = _ExtraOptions() and _SSLOptions = _SSLOptions(). Then the optional goes away because the variable is always set, everywhere.

Reply:
> I had _SSLOptions set as default to _SSLOptions() before and you asked me to remove it and have it default to None. If options.ssl can be None, it makes sense that the function can accept None. If you don't want it to accept None, I think the most reasonable way to accomplish it is to have _ExtraOptions = _ExtraOptions() and _SSLOptions = _SSLOptions(). Then the optional goes away because the variable is always set, everywhere.

You updated the caller of this helper function exactly as I told you which 

> We always want options set in this function.

related to. With the recent change you made options is no longer optional in this helper and is always set, so the signature needs to be updated and the check dropped.